api/migrations/versions/72e6af46be22_add_scope_column_to_user_group_table.py

```diff
@@ -0,0 +1,73 @@
+"""Add scope column to user group table
+
+Revision ID: 72e6af46be22
+Revises: 8c480b641f22
+Create Date: 2023-11-06 12:23:06.476645
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "72e6af46be22"
+down_revision = "8c480b641f22"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column("user_group", sa.Column("scopes", sa.String, nullable=True))
+    # Add scopes for default roles
+    separator = ","
+    scopes_everyone_has = "task:create-own,task:read-own,task:update-own,task:delete-own,task_type:read,template:create-own,template:update-own,template:read-own,template:read-global,template:delete-own,project:read-assigned,vacation:read-own,vacation:read-other,vacation:create-own,vacation:update-own,vacation:delete-own"
+    hr_only_scopes = "long_leave:create,long_leave:update,long_leave:delete,user_capacity:create,user_capacity:update,user_capacity:delete"
+    hr_scopes = separator.join((scopes_everyone_has, hr_only_scopes))
+    pm_only_scopes = "project:create,project:update-own,project:delete-own,project:read-own,project:read-assigned,project:read-other,client:create,client:update,client:delete,client:read,project_status:read"
+    pm_scopes = separator.join((scopes_everyone_has, pm_only_scopes))
+    manager_only_scopes = (
+        "template:create-global,template:update-global,template:delete-global"
+    )
+    manager_scopes = separator.join(
+        (scopes_everyone_has, hr_only_scopes, pm_only_scopes, manager_only_scopes)
+    )
+    admin_only_scopes = "task_type:create,task_type:update,task_type:delete,config:update,project_status:create,project_status:update,project_status:delete"
+    admin_scopes = separator.join(
+        (
+            scopes_everyone_has,
+            pm_only_scopes,
+            hr_only_scopes,
+            manager_only_scopes,
+            admin_only_scopes,
+        )
+    )
+    conn = op.get_bind()
+    staff_query = sa.text(
+        f"UPDATE user_group SET scopes = '{scopes_everyone_has}' WHERE name = 'staff'"
+    )
+    conn.execute(staff_query)
+    pm_query = sa.text(
+        f"UPDATE user_group SET scopes = '{pm_scopes}' WHERE name = 'project manager'"
+    )
+    conn.execute(pm_query)
+    hr_query = sa.text(
+        f"UPDATE user_group SET scopes = '{hr_scopes}' WHERE name = 'human resources'"
+    )
+    conn.execute(hr_query)
+    manager_query = sa.text(
+        f"UPDATE user_group SET scopes = '{manager_scopes}' WHERE name = 'manager'"
+    )
+    conn.execute(manager_query)
+    admin_query = sa.text(
+        f"UPDATE user_group SET scopes = '{admin_scopes}' WHERE name = 'admin'"
+    )
```

Comment:
These groups don't exist yet, so maybe we need to create them in this migration as well?

Reply:
Good point. I think we have a note in the dev docs about how to add some groups with a sql statement, but nothing in any of the migrations. I will add this to the migration.